### PR TITLE
Add helper function for finding UARTs and getting their configurations

### PR DESCRIPTION
**Pull request by fhunleth**
I had an issue where I was doing a lot of Nerves.UART work via the iex prompt on a device. I started losing my UART pids and wasn't sure which UARTs to kill. I'm not sure that this is a perfect way of solving this problem, but it seems to help. Also adding a function to get the current UART's configuration seemed like a natural API to have even though I've gotten along without it for a long time.

### CODE REVIEW

File: lib/nerves_uart.ex [hunk 1]
```diff
@@ -173,6 +202,14 @@ defmodule Nerves.UART do
     GenServer.call(pid, {:configure, opts})
   end
 
+  @doc """
+  Get the configuration of the serial port.
+  """
+  @spec configuration(GenServer.server()) :: {binary, [uart_option]}
```
Comment (tmecklem) [inline lib/nerves_uart.ex]: Is this part intended to be used in production or also just for diagnostic purposes like `find_started_uarts` above?

Comment (fhunleth) [reply lib/nerves_uart.ex]: It started out as only for `find_started_uarts`, but then it seemed useful.

File: lib/nerves_uart.ex [hunk 2]
```diff
@@ -74,6 +74,35 @@ defmodule Nerves.UART do
...
+  updates to the interface make it more convenient to use.
+  """
+  @spec find_started_uarts() :: [{binary, pid()}]
+  def find_started_uarts() do
+    Process.list()
+    |> Enum.filter(&is_nerves_uart_process/1)
+    |> Enum.map(&nerves_uart_info/1)
+  end
+
+  defp is_nerves_uart_process(pid) do
```
Comment (electricshaman) [inline lib/nerves_uart.ex]: Clever what you did here.  How did you find this solution?

Comment (fhunleth) [reply lib/nerves_uart.ex]: `:observer`

Comment (electricshaman) [reply lib/nerves_uart.ex]: Nice.  Part of me wonders if we should find a 2nd way of doing this since it's part of OTP's internal API.

**Pull request merged**